src/Parsers/ExpressionListParsers.cpp

```diff
@@ -599,158 +609,1974 @@ bool ParserLambdaExpression::parseImpl(Pos & pos, ASTPtr & node, Expected & expe
 }
 
 
-bool ParserTableFunctionExpression::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+ASTPtr makeBetweenOperator(bool negative, ASTs arguments)
 {
-    if (ParserTableFunctionView().parse(pos, node, expected))
-        return true;
-    return elem_parser.parse(pos, node, expected);
+    // subject = arguments[0], left = arguments[1], right = arguments[2]
+    auto f_combined_expression = std::make_shared<ASTFunction>();
+    auto args_combined_expression = std::make_shared<ASTExpressionList>();
+
+    /// [NOT] BETWEEN left AND right
+    auto f_left_expr = std::make_shared<ASTFunction>();
+    auto args_left_expr = std::make_shared<ASTExpressionList>();
+
+    auto f_right_expr = std::make_shared<ASTFunction>();
+    auto args_right_expr = std::make_shared<ASTExpressionList>();
+
+    args_left_expr->children.emplace_back(arguments[0]);
+    args_left_expr->children.emplace_back(arguments[1]);
+
+    args_right_expr->children.emplace_back(arguments[0]);
+    args_right_expr->children.emplace_back(arguments[2]);
+
+    if (negative)
+    {
+        /// NOT BETWEEN
+        f_left_expr->name = "less";
+        f_right_expr->name = "greater";
+        f_combined_expression->name = "or";
+    }
+    else
+    {
+        /// BETWEEN
+        f_left_expr->name = "greaterOrEquals";
+        f_right_expr->name = "lessOrEquals";
+        f_combined_expression->name = "and";
+    }
+
+    f_left_expr->arguments = args_left_expr;
+    f_left_expr->children.emplace_back(f_left_expr->arguments);
+
+    f_right_expr->arguments = args_right_expr;
+    f_right_expr->children.emplace_back(f_right_expr->arguments);
+
+    args_combined_expression->children.emplace_back(f_left_expr);
+    args_combined_expression->children.emplace_back(f_right_expr);
+
+    f_combined_expression->arguments = args_combined_expression;
+    f_combined_expression->children.emplace_back(f_combined_expression->arguments);
+
+    return f_combined_expression;
 }
 
+enum Action
+{
+    OPERAND,
+    OPERATOR
+};
 
-bool ParserPrefixUnaryOperatorExpression::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+enum OperatorType
```

Comment:
The comment is needed here.
If I want to add a new operator, which type should I choose?

Reply:
Types are for us to add specific handling of these operators. So in case when the new operator does not need any special handling - it does not need to be assigned any type.